doc/design/scope.md

```diff
@@ -0,0 +1,58 @@
+# Scope
+
+### Define
+
+Scope is a context to manage Variables. It mainly contains a map from Variable name to Variable. Net will get and update variable throw scope.
+
+```cpp
+class Variable;
+using VariablePtr = std::shared_ptr<Variable>;
+
+class Scope final {
+public:
```

Comment:
Just a reminder -- I've put .clang-format files that strictly follows Google style in new source directories `paddle/platform`, `paddle/framework`.  So this line would trigger clang-format error because Google style requires a single space before `public:`.